src/silx/gui/plot/tools/profile/manager.py

```diff
@@ -54,75 +55,41 @@
 class _RunnableComputeProfile(qt.QRunnable):
     """Runner to process profiles
 
-    :param qt.QThreadPool threadPool: The thread which will be used to
-        execute this runner. It is used to update the used signals
-    :param ~silx.gui.plot.items.Item item: Item in which the profile is
-        computed
-    :param ~silx.gui.plot.tools.profile.core.ProfileRoiMixIn roi: ROI
-        defining the profile shape and other characteristics
+    :param item: Item in which the profile is computed
+    :param roi: ROI defining the profile shape and other characteristics
     """
 
-    class _Signals(qt.QObject):
+    class Signal(qt.QObject):
         """Signal holder"""
-
-        resultReady = qt.Signal(object, object)
         runnerFinished = qt.Signal(object)
 
-    def __init__(self, threadPool, item, roi):
-        """Constructor"""
+    def __init__(self, item: items.Item, roi: ProfileRoiMixIn):
         super(_RunnableComputeProfile, self).__init__()
-        self._signals = self._Signals()
-        self._signals.moveToThread(threadPool.thread())
```

Comment:
Maybe better to move it into another PR.
That such a thing which is difficult to diagnose in case of trouble

Reply:
I removed this change from this PR and opened PR #4169 for it.